quarkus/config-api/src/main/java/org/keycloak/config/FeatureOptions.java

```diff
@@ -6,29 +6,35 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class FeatureOptions {
 
     public static final Option<List> FEATURES = new OptionBuilder("features", List.class, Profile.Feature.class)
             .category(OptionCategory.FEATURE)
             .description("Enables a set of one or more features.")
-            .expectedValues(FeatureOptions::getFeatureValues)
             .defaultValue(Optional.empty())
+            .expectedValues(() -> getFeatureValues(true))
             .buildTime(true)
             .build();
 
     public static final Option FEATURES_DISABLED = new OptionBuilder("features-disabled", List.class, Profile.Feature.class)
             .category(OptionCategory.FEATURE)
             .description("Disables a set of one or more features.")
-            .expectedValues(FeatureOptions::getFeatureValues)
+            .expectedValues(() -> getFeatureValues(false))
             .buildTime(true)
             .build();
 
-    private static List<String> getFeatureValues() {
+    public static List<String> getFeatureValues(boolean includeVersions) {
         List<String> features = new ArrayList<>();
 
-        for (Profile.Feature value : Profile.Feature.values()) {
-            features.add(value.getKey());
+        if (includeVersions) {
+            Profile.getAllUnversionedFeatureNames().forEach(f -> {
```

Comment:
Would not be better to provide sth like `Profile.getAllVersionedFeatureNames()` to have the approach centralized?

Reply:
I'm not sure I entirely follow you. This method is returning values that are specific to the options, so it belongs with the FeatureOptions.